plugins/zoom-to-fit/src/index.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview UI control for zoom-to-fit.
+ */
+
+import Blockly from 'blockly/core';
+
+/**
+ * Class for zoom to fit control.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+ * @implements {Blockly.IPositionable}
+ * @constructor
+ */
+export class ZoomToFitControl {
+  /**
+   * Constructor for zoom-to-fit control.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the plugin will
+   *     be added to.
+   */
+  constructor(workspace) {
+    /**
+     * The workspace.
+     * @type {!Blockly.WorkspaceSvg}
+     * @protected
+     */
+    this.workspace_ = workspace;
+
+    /**
+     * The SVG group containing the zoom controls.
+     * @type {SVGElement}
+     * @private
+     */
+    this.svgGroup_ = null;
+
+    /**
```

Comment:
I think storing the top/left/width/height properties separately is more beneficial than storing them all on a rectangle for the following reasons:
1. When returning the bounding box for the object, there's no concern that the caller could modify the properties on the bounding box (which could mess up other computation during positioning) since it is a new object every time.
2. I'd have to re-compute the width/height every time `position` is called to use for computing the new bounding box.
2. (minor plus) It allows me to conveniently call `getBoundingBox` during `position` logic to quickly recompute the new potential bounding box

Reply:
For the first one: definitely would want to return a copy, but that's easy.

For the second and third ones: okay, fine to leave it this way since it has these benefits.